src/components/catalog-search/catalog-search.controller.ts

```diff
@@ -19,12 +20,13 @@ export class CatalogSearchController implements angular.IController {
   // Used when the user starts typing before the items have loaded.
   private searchDeferred: ng.IDeferred<any[]>;
 
-  constructor($rootScope: any, $scope: any, $timeout: any, $q: any, Catalog: any, KeywordService: any) {
+  constructor($rootScope: any, $scope: any, $timeout: any, $q: any, Catalog: any, Constants: any, KeywordService: any) {
```

Comment:
I think the basic problem here and elsewhere is KeywordService comes from common, which is not Typescript. So we don't have the interface (except for the mock service, which is incomplete and also unfortunately duplicates a lot of the same code from common).

I agree we're not using types well. This is a general problem in the catalog repo. We have #593 to track.

IMO, we should decide if we really want Typescript or not and go all-in. We're in a weird space in between where we're not seeing the advantages. We're at an inflection point where we could look at doing that since we're talking about moving to React.